src/Data/LLVM/BitCode/Record.hs

```diff
@@ -182,5 +182,10 @@ oldOrStrtabName n r = do
           Just st -> do
             offset <- parseField r 0 numeric
             len <- parseField r 1 numeric
-            return (resolveStrtabSymbol st offset len, 2)
+            if len == 0
+            then do
+              n <- show <$> nextSymbolId
+              return (Symbol n, 2)
```

Comment:
Can you leave some comments explaining what is going on in the `len == 0` case? From [this commit](https://github.com/GaloisInc/llvm-pretty-bc-parser/commit/c56e3da2124fddc1c249a765c6cb304299bcf8b0), I gather that it related to unnamed symbols in some way, but the details aren't clear to me.